chia/data_layer/data_layer_wallet.py

```diff
@@ -1251,9 +1251,9 @@ async def finish_graftroot_solutions(offer: Offer, solver: Solver) -> Offer:
         for spend in offer.coin_spends():
             matched, curried_args = match_dl_singleton(spend.puzzle_reveal.to_program())
             if matched and spend.coin.name() not in all_parent_ids:
-                innerpuz, root, launcher_id = curried_args
+                innerpuz, root_prg, launcher_id = curried_args
                 singleton_struct = launcher_to_struct(bytes32(launcher_id.as_python())).get_tree_hash()
-                singleton_to_root[singleton_struct] = bytes32(root.as_python())
+                singleton_to_root[singleton_struct] = bytes32(root_prg.as_python())
```

Comment:
this change warrants some scrutiny. the variable `root` is used below as well (but with the type `str`). As far as I can tell, this version of `root` was intended to be isolated to this scope.